src/theme/DocItem/Layout/index.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import clsx from 'clsx';
+
+import { useWindowSize } from '@docusaurus/theme-common';
+import { useDoc } from '@docusaurus/theme-common/internal';
+import DocItemPaginator from '@theme/DocItem/Paginator';
+import DocVersionBanner from '@theme/DocVersionBanner';
+import DocVersionBadge from '@theme/DocVersionBadge';
+import DocItemFooter from '@theme/DocItem/Footer';
+import DocItemTOCMobile from '@theme/DocItem/TOC/Mobile';
+import DocItemTOCDesktop from '@theme/DocItem/TOC/Desktop';
+import DocItemContent from '@theme/DocItem/Content';
+import DocBreadcrumbs from '@theme/DocBreadcrumbs';
+
+import useToc from '@site/src/utils/useToc';
+
+import styles from './styles.module.css';
+/**
+ * Decide if the toc should be rendered, on mobile or desktop viewports
+ */
+function useDocTOC() {
+  const { frontMatter } = useDoc();
+  const toc = useToc();
```

Comment:
This is an actual change. We want to use our ("correct") TOC to detect its size. Previously, the condition below (`const canRender = !hidden && toc.length > 0;`) was based on TOC by Docusaurus. In the case of the page composed only of components - TOC was always empty.